builder/dnf_control.go

```diff
@@ -0,0 +1,330 @@
+// Copyright © 2018 Intel Corporation
```

Comment:
I was trying to keep the functional changes entirely out of this PR since it is so big (moving this to the `config` package will require changing the callers), but I guess I can do those changes as another commit so that commit can be reviewed more carefully.

Reply:
Ugh, I commented before I tried doing actually doing it. Issues I immediately see:
* almost all the functions have a `builder.Builder` receiver
* can't import `builder` from `config` because `import cycle not allowed` so I can't just change the receiver to be an argument to each function. This would actually require a slight (significant?) refactor because of all the internals of the `Builder` object these functions use and touch.

So not doing this in this PR.

Reply:
Fair enough.